pub_key.proto

```diff
@@ -24,26 +24,60 @@ message PubKeyMethod {
 }
 
 message NewPubKeyPayload {
-    enum EntityType {
-        PERSONAL = 0;
-        SERVER = 1;
+    enum PublicKeyType {
+        RSA = 0;
+        ECDSA = 1;
+        EDDSA = 2;
     }
 
-    enum PubKeyType {
-        RSA = 0;
+    enum HashingAlgorithm {
+        SHA256 = 0;
+        SHA512 = 1;
+    }
+
+    message RSAConfiguration {
+        enum Padding {
+            PSS = 0;
+            PKCS = 1;
+        }
+
+        uint64 modulus = 1;
+        bytes exponent = 2;
+        Padding padding = 3;
+
+        uint32 valid_from = 4;
+        uint32 valid_to = 5;
+    }
+
+    message ECDSAConfiguration {
+        enum EC {
+            SECP256K1 = 0;
+        }
+
+        bytes key = 1;
+        EC ec = 2;
+    }
+
+    message EdDSAConfiguration {
+        enum EC {
+            Curve25519 = 0;
```

Comment:
Curve25519 is DH key exchange algorithm not algorithm for digital signature, use ED25519 instead

Reply:
https://en.wikipedia.org/wiki/EdDSA#Ed25519